src/backend/joanie/core/serializers/client.py

```diff
@@ -873,3 +873,71 @@ def get_abilities(self, user) -> dict:
         if request:
             return request.user.get_abilities(user)
         return {}
+
+
+class GenerateSignedContractsZipSerializer(serializers.Serializer):
+    """
+    Serializer used by both view and command generating a zip containing signed contracts
+    """
+
+    course_product_relation_id = serializers.UUIDField(allow_null=True, required=False)
+    organization_id = serializers.UUIDField(allow_null=True, required=False)
+
+    def validate(self, attrs):
+        """
+        Validate that course_product_relation_id and organization_id are mutually exclusive
+        but at least one is required.
+
+        Also, it fetch in database the corresponding object to add them in the validated data.
+        """
+        if attrs.get("course_product_relation_id") and attrs.get("organization_id"):
+            raise serializers.ValidationError(
+                "You must set exactly one parameter for the method. It cannot be both. "
+                "You must choose between an Organization UUID or a Course Product Relation UUID."
+            )
+
+        if not attrs.get("course_product_relation_id") and not attrs.get(
+            "organization_id"
+        ):
+            raise serializers.ValidationError(
+                "You must set at least one parameter for the method."
+                "You must choose between an Organization UUID or a Course Product Relation UUID."
+            )
+
+        errors = {}
+        if attrs.get("course_product_relation_id"):
+            try:
+                attrs[
+                    "course_product_relation"
+                ] = models.CourseProductRelation.objects.get(
+                    pk=attrs.get("course_product_relation_id")
+                )
+            except models.CourseProductRelation.DoesNotExist:
+                errors["course_product_relation_id"] = (
+                    "Make sure to give an existing course product relation UUID. "
+                    "No CourseProductRelation was found with the given "
+                    f"UUID : {attrs.get('course_product_relation_id')}."
+                )
+
+        if attrs.get("organization_id"):
+            try:
+                attrs["organization"] = models.Organization.objects.get(
+                    pk=attrs.get("organization_id")
+                )
```

Comment:
```suggestion
        course_product_relation_id = attrs.get("course_product_relation_id")
        organization_id = attrs.get("organization_id")
        if course_product_relation_id and organization_id:
            raise serializers.ValidationError(
                "You must set exactly one parameter for the method. It cannot be both. "
                "You must choose between an Organization UUID or a Course Product Relation UUID."
            )

        if not course_product_relation_id and not organization_id:
            raise serializers.ValidationError(
                "You must set at least one parameter for the method."
                "You must choose between an Organization UUID or a Course Product Relation UUID."
            )

        errors = {}
        if course_product_relation_id:
            try:
                attrs[
                    "course_product_relation"
                ] = models.CourseProductRelation.objects.get(
                    pk=course_product_relation_id
                )
            except models.CourseProductRelation.DoesNotExist:
                errors["course_product_relation_id"] = (
                    "Make sure to give an existing course product relation UUID. "
                    "No CourseProductRelation was found with the given "
                    f"UUID : {attrs.get('course_product_relation_id')}."
                )

        if organization_id:
            try:
                attrs["organization"] = models.Organization.objects.get(
                    pk=organization_id
                )
```

Reply:
What is your motivation for this change ?

Reply:
just to remove `attrs.get("…` everywhere. Cleaner to me.